test/extended/apiserver/latency.go

```diff
@@ -0,0 +1,101 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	promhelper "github.com/openshift/origin/test/extended/util/prometheus"
+)
+
+// TODO: remove these variables once we move to route instead of exec pod
+var (
+	oc                   *exutil.CLI
+	url, bearerToken, ns string
+	execPod              *v1.Pod
+)
+
+var _ = g.Describe("[sig-instrumentation][Late] API Server latency", func() {
+	defer g.GinkgoRecover()
+
+	oc = exutil.NewCLIWithoutNamespace("apiserver-latency")
+
+	g.BeforeEach(func() {
+		err := exutil.WaitForAnImageStream(
+			oc.AdminImageClient().ImageV1().ImageStreams("openshift"), "tools",
+			exutil.CheckImageStreamLatestTagPopulated, exutil.CheckImageStreamTagNotFound,
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		var ok bool
+		url, _, bearerToken, ok = promhelper.LocatePrometheus(oc)
+		if !ok {
+			e2e.Failf("Prometheus could not be located on this cluster, failing prometheus test")
+		}
+	})
+
+	// Lantecy thresholds collected from:
+	// - https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.9-e2e-gcp-upgrade/1480678513441771520
+	// - https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.9-e2e-gcp/1479392960888967168
+	// - https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.9-e2e-aws-serial/1479392960926715904
+	g.It("should verify that the apiserver requests latency are within expected thresholds", func() {
+		ns = oc.SetupNamespace()
+
+		// TODO: access Prometheus via route instead of exec pod
+		execPod = exutil.CreateExecPodOrFail(oc.AdminKubeClient(), ns, "execpod")
+		defer func() {
+			oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+		}()
+
+		// 99th percentile of resource-scoped requests latency
+		scope := "resource|"
+		expectRequestLatency("GET", scope, parseDuration("75ms"))
```

Comment:
can you indicate the actuals you got so that when this later fails and Abu or Stefan say, "this isn't significantly worse", I can say, "yes, it is significantly worse, see this historical value from 1/15/2022".

Reply:
done